metricbeat/module/kubernetes/container/container.go

```diff
@@ -71,25 +73,33 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}, nil
 }
 
-// Fetch methods implements the data gathering and data conversion to the right format
-// It returns the event which is then forward to the output. In case of an error, a
-// descriptive error must be returned.
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 	m.enricher.Start()
 
 	body, err := m.http.FetchContent()
 	if err != nil {
-		return nil, err
+		logger.Error(err)
```

Comment:
similar here

Reply:
Here I really think it is better to give some more context like "error doing HTTP request to fetch container metrics". Good catch!

Reply:
Thanks :-)